apps/evm/lib/evm/operation/system.ex

```diff
@@ -97,7 +97,7 @@ defmodule EVM.Operation.System do
     sub_state =
       n_sub_state
       |> SubState.merge(sub_state)
-      |> SubState.add_touched_account(result)
+      |> SubState.add_touched_account(exec_env.address)
```

Comment:
Why did you change touched account here?

Reply:
The SubState's touched accounts should have the binary format for the address of an account. This operation was passing the `result` which is an unsigned integer, when it should have been passing the `exec_env.address` instead.

Reply:
but we should add to touched accounts a created address, not address from exec env

Reply:
Oh that's right. We need to change this to the created address, but it should be the binary format. I know there's a similar `Contract.Address.new` in the blockchain that does the same thing. We really should have one way of creating new addresses, but we'll look into that. Good catch!

Reply:
We updated this in https://github.com/poanetwork/mana/pull/423/commits/8872f8fb557939d9ec1788952d3f520344dc7632